src/main/java/Duke.java

```diff
@@ -1,10 +1,148 @@
+    }
+
+    public static void runDuke() {
+        tasks = store.getStoredTasks();
+        numTasks = tasks.size();
+        greeting();
+
+        Scanner sc = new Scanner(System.in);
+
+        while (true) {
+            String input = sc.nextLine();
+
+            try {
+                String command = getCommand(input);
+                if (command.equals("bye")) {
+                    store.update(tasks);
+                    break;
+                }
+                doInstructions(input);
+            } catch (DukeException e) {
+                System.out.printf("Error: %s\n\n", e.getMessage());
+            }
+        }
+        System.out.println("Bye! Hope you don't come back!");
+    }
+
+    public static void doInstructions(String input) throws DukeException {
+//        Task[] list = new Task[100];
+        String command = getCommand(input);
+
+        if (command.equals("list")) {
+            printList(numTasks);
+        } else if (command.equals("done")) {
+            int taskNum = Integer.parseInt(input.split(" ")[1]);
+            // check for
+            // 1. invalid task num
+            // 2. empty task num
+            // 3. if task is already done?
+            // implement ui class? if have time to print out done statement
+            tasks.get(taskNum - 1).markAsDone();
+            System.out.println(delimiter());
+            System.out.println("I've finally done this task:");
+            System.out.println(tasks.get(taskNum - 1).toString());
+            System.out.println(delimiter() + "\n");
+        } else if (command.equals("delete")) {
+            int taskNum = Integer.parseInt(input.split(" ")[1]);
+            String desc = tasks.get(taskNum - 1).toString();
+            tasks.remove(taskNum - 1);
+            numTasks -= 1;
+            System.out.println(delimiter());
+            System.out.println("Do you really want to remove this? Fine. I've removed:\n    " + desc);
+            System.out.printf("Now you have %d tasks in your list\n", numTasks);
+            System.out.println(delimiter() + "\n");
+        } else {
+            Task newTask;
+            // check for
+            // 1. no description
+            // 2. no time
+            if (command.equals("todo")) {
+                newTask = new ToDo(getDescription(input));
+            } else if (command.equals("deadline")) {
+                newTask = new Deadline(getDescription(input), getTime(input));
+            } else { // (command.equals("event")) {
+                newTask = new Event(getDescription(input), getTime(input));
+            }
+
+            tasks.add(newTask);
+            numTasks += 1;
+            System.out.println(delimiter());
+            System.out.println("Alrighty, you added:");
+            System.out.println("  " + newTask.toString());
+            System.out.printf("Now you have %d tasks in your list.\n", numTasks);
+            System.out.println(delimiter() + "\n");
```

Comment:
Can respond to user in a more systematic way by putting all the responding mechanism (those with System.out.println ) to a new class.